lib/unleash/feature_toggle.rb

```diff
@@ -75,9 +76,47 @@ def am_enabled?(context)
       evaluate(context).enabled?
     end
 
+    def parent_dependencies_satisfied?(context)
+      return true if dependencies.empty?
+      return dependencies.all? do |parent|
+        evaluate_parenthood(parent, context)
+      end
+    end
+
+    def evaluate_parenthood(parent, context)
+      parent_toggle = get_parent(parent["feature"])
+      if parent_toggle.nil?
+        return false
+      end
+      return false unless parent_toggle.dependencies.empty?
+      evaluation_result = parent_toggle.is_enabled?(context)
+      if parent["enabled"] == false
+        return !evaluation_result
+      else
+        return false unless evaluation_result
+      end
+
+      unless parent["variants"].nil? || parent["variants"].empty?
+        return parent["variants"].include?(parent_toggle.get_variant(context).name)
+      end
+      return evaluation_result
+    end
+
+    def get_parent(feature)
+      toggle_as_hash = Unleash&.toggles&.select{ |toggle| toggle['name'] == feature }&.first
```

Comment:
is there Array.find method that performs select and first?

Reply:
yes

Reply:
fixed